pkg/proc/gdbserial/gdbserver.go

```diff
@@ -698,6 +698,7 @@ continueLoop:
 		// in backward continue mode.
 		case 0:
 			if p.conn.direction == proc.Backward {
+				p.ClearInternalBreakpoints()
```

Comment:
There's no way for the caller to know that this happened, i.e. that we reached the start of the recording. So it can't be done there. We could return a boolean specifically for this.

Reply:
Yeah I think we will have to rethink this as we move breakpoint handling up to Target.

Reply:
Is there any reason not to clear all internal breakpoints when proc.Continue returns? I realize a next or step operation could be interrupted by a manual stop, or hitting another breakpoint and so forth, but in each case wouldn't we want to clear that internal breakpoint state since the step/next operation would be cancelled?

Reply:
No, we are giving the option to the frontend to either cancel the next operation or to execute another continue and finish it. Both the command line frontend and gdlv are taking advantage of this.

Reply:
Ack, you're right.

Reply:
I think we should communicate this back to the caller (e.g. Target) via a boolean or an error similar to `ErrProcessExited`.

Reply:
I've added a new field to Target that I think will also be useful for dap and used the same mechanism to implement this (#1877)